### ISSUE TITLE
maxArrayLens order.nodes, state.nodes and base.node

### ISSUE THREAD

**Issue by HardyBenjamin** [VDA5050/VDA5050#298]
Hi,
I have a number of questions, but they are related.

In the factsheet, the parameter "order.nodes" in "maxArrayLens" is the max number of nodes allowed in an order message ? The parameter "state.nodeStates" is the max number of nodes in the node base ?

When an AGV receives a new order, does it create a new base (populate). When an AGV receives an update order, does it append the new nodes to the old ones (append). But when the AGV passes nodes, sould/can those nodes be deleted from the base ? Does this action release space in the array for storing the new nodes in the new update order, or does it involve moving array elements to the beginning (leaving space at the end of the table for new nodes) ? Or is the base node array (order node array) is not dynamic ?

Example :
If the AGV have an array of 6 nodes and can receive 3 nodes per order update, the master controller can send 1 order and 1 update with 3 nodes each. And when the AGV has passed 2 nodes, master controller can send a new update with 2 nodes ? Or do master controller need to wait until the AGV has reached the last node of the first update before sending the new update?

TY.

**Comment by collani-bosch** [comment]
Good point, maybe we should specify more precisely the definition of this fields. I would say "order.nodes" specifies the maximum number of nodes inside an order at all. "state.nodeStates" is also for me a little bit confusing.
In which way you implement these protocol requirements is up to you, because the VDA5050 is protocol specification and not a software design guide. :-) The VDA5050 defines that traversed nodes and edges are not longer included inside the state message. This doesn't mean that the vehicle shall throw away these information. If the information is still useful keep it, but do not include them inside the state message.
